lib/connection.js

```diff
@@ -405,18 +400,15 @@ Connection.prototype._tryReceiveHeader = function(header) {
 };
 
 Connection.prototype._tryReceiveFrame = function() {
-  try {
-    var frame = FrameReader.read(this._buffer);
-    if (frame === undefined) {
+  var frame = frames.readFrame(this._buffer);
```

Comment:
Honestly, this method `_tryReceiveFrame` is spurious and can be deleted in favor of just calling `frames.readFrame`